src/test/ui/async-await/async-fn-nonsend.rs

```diff
@@ -0,0 +1,59 @@
+// compile-fail
+// edition:2018
+// compile-flags: --crate-type lib
+
+#![feature(async_await)]
+
+use std::{
+    cell::RefCell,
+    fmt::Debug,
+    rc::Rc,
+};
+
+fn non_sync() -> impl Debug { RefCell::new(()) }
+
+fn non_send() -> impl Debug { Rc::new(()) }
+
+fn take_ref<T>(_: &T) {}
+
+async fn fut() {}
+
+async fn fut_arg<T>(_: T) {}
+
+async fn local_dropped_before_await() {
+    // FIXME: it'd be nice for this to work.
```

Comment:
Can you clarify this note wrt. not being `Send` because you could get the impression from this note that you are saying the function itself doesn't compile (which it does, https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=85207a519d05dba162b213519b02bf11)

Reply:
done